### PR TITLE
bug fix for lookup leak when we remove the last lookup from lookup tier

### PR DESCRIPTION
**Pull request by FaxianZhao**
Fixes bug which caused by #7852.

#### Fixed the bug ...
Due to #7852, coordinator won't tell other node to dispose lookup when we remove the last lookup from a tier.
#### Added a forbidden-apis entry ...

<hr>

This PR has:
- [x] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/incubator-druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

<hr>

##### Key changed/added classes in this PR
 * `LookupCoordinatorManager`

### CODE REVIEW

**Comment by himanshug** [comment]
@FaxianZhao thanks for discovering and finding the root cause.

File: server/src/main/java/org/apache/druid/server/lookup/cache/LookupCoordinatorManager.java [hunk 1]
```diff
@@ -557,16 +557,16 @@ void lookupManagementLoop()
 
     try {
       List<ListenableFuture<Map.Entry>> futures = new ArrayList<>();
-      for (Map.Entry<String, Map<String, LookupExtractorFactoryMapContainer>> tierEntry : allLookupTiers.entrySet()) {
+      for (String tier : lookupNodeDiscovery.getAllTiers()) {
```
Comment (himanshug) [inline server/src/main/java/org/apache/druid/server/lookup/cache/LookupCoordinatorManager.java]: can you merge https://github.com/FaxianZhao/incubator-druid/pull/1 to add another block to note lookups that will never be loaded?

Comment (FaxianZhao) [reply server/src/main/java/org/apache/druid/server/lookup/cache/LookupCoordinatorManager.java]: > can you merge [FaxianZhao#1](https://github.com/FaxianZhao/incubator-druid/pull/1) to add another block to note lookups that will never be loaded?

Good idea.
But for the users who enable 'druid.lookup.lookupTierIsDatasource', will this warning confuse them?

Comment (himanshug) [reply server/src/main/java/org/apache/druid/server/lookup/cache/LookupCoordinatorManager.java]: a quick look at the usage of that property tells me that it is not used by coordinator and shouldn't interfere.

Comment (FaxianZhao) [reply server/src/main/java/org/apache/druid/server/lookup/cache/LookupCoordinatorManager.java]: > a quick look at the usage of that property tells me that it is not used by coordinator and shouldn't interfere.

Agree.

**Pull request merged**